front/app/components/PostShowComponents/Comments/CommentVote.tsx

```diff
@@ -128,104 +126,82 @@ interface InputProps {
 
 interface DataProps {
   commentVotingPermissionInitiative: GetInitiativesPermissionsChildProps;
-  authUser: GetAuthUserChildProps;
   comment: GetCommentChildProps;
-  commentVote: GetCommentVoteChildProps;
 }
 
 interface Props extends InputProps, DataProps {}
 
 const CommentVote = ({
   comment,
-  commentVote,
   postId,
   postType,
   commentId,
   commentType,
-  authUser,
   commentVotingPermissionInitiative,
   className,
   intl: { formatMessage },
 }: Props & WrappedComponentProps) => {
-  const [voted, setVoted] = useState(false);
-  const [upvoteCount, setUpvoteCount] = useState(0);
+  const authUser = useAuthUser();
+  const { mutate: deleteCommentVote } = useDeleteCommentVote();
+  const { mutate: addCommentVote } = useAddCommentVote();
+  const { data: commentVote } = useCommentVote(
+    !isNilOrError(comment)
+      ? comment?.relationships?.user_vote?.data?.id
+      : undefined
```

Comment:
You could pass comment via the parent, it's available there. Then you can remove GetComment and we have a hook for GetInitiativesPermissions as well, so you could even get rid of all data loading via react-adopt.

And you could remove a bunch of isNilOrError checks for comment. :)